streams/src/main/java/org/apache/kafka/streams/processor/internals/Tasks.java

```diff
@@ -65,13 +75,10 @@ class Tasks {
     private Consumer<byte[], byte[]> mainConsumer;
 
     Tasks(final LogContext logContext,
-          final TopologyMetadata topologyMetadata,
```

Comment:
This is for case 2) in description: we detect unknown named topologies in the caller (`TaskManager`) so `Tasks` would be just dumb book-keeping.